ember/app/ui/cases/detail/index/controller.js

```diff
@@ -25,6 +26,50 @@ export default class CasesDetailIndexController extends Controller {
     return this.model.workItems.filterBy("status", "READY").length;
   }
 
+  get remarks() {
+    return this.model.workItems
+      .reduce((workItems, workItem) => {
+        if (
+          !Object.keys(ENV.APP.caluma.displayedAnswers).includes(
+            workItem.task.slug
+          )
+        ) {
+          return workItems;
+        } else if (!workItems.length) {
+          return [...workItems, workItem];
+        }
+        const duplicateIndex = workItems.findIndex(
+          (item) => item.task.slug === workItem.task.slug
+        );
+
+        if (duplicateIndex === -1) {
+          return [...workItems, workItem];
+        } else if (
+          new Date(workItems[duplicateIndex].createdAt) <
+          new Date(workItem.createdAt)
+        ) {
+          workItems.splice(duplicateIndex, 1);
+          return [...workItems, workItem];
+        }
+
```

Comment:
I think all of this can be done outside of a `reduced` or not?

Reply:
The `reduce` is used to filter for the correct workItems, it also needs to filter out older workItems of the same task. I'm not sure how you would do it without a loop. `filter` doesn't work as we need the context of which workItems are older than the other.